### ISSUE TITLE
What is the replacement for this.$children?

### ISSUE THREAD

**Issue by ddenev** [vuejs/composition-api#411]
What would be the official replacement for `this.$children` in the new composition API?

I need to access the child components/elements of a component.

**Comment by posva** [comment]
I think it's the `vnode` property when getting the instance through `getCurrentInstance` (https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/component.ts#L164)

I don't remember having any previous discussion about it though

**Comment by ddenev** [comment]
@antfu, @posva, thank you for the swift reply!

`getCurrentInstance().$vnode.children` is `undefined` even though there are children nodes:
```
<my-component>
  <other-component v-for="el of list" :key="el.id" />
</my-component>
```

Also it turns out that `getCurrentInstance().$children` gives me only components (no elements). So actually I need children VNodes, and not `$children` (sorry about the confusion). I guess that would be the official replacement, right?

For my case I found that I can use `context.slots.default()` to get the direct children VNodes (works for both elements and components).

**Comment by antfu** [comment]
@ddenev did you try to wrap `getCurrentInstance().$children` inside `onMounted`?

**Comment by ddenev** [comment]
If you are referring to my '_Also it turns out that `getCurrentInstance().$children` gives me only components (no elements)_' comment above then yes - `getCurrentInstance().$children` within `onMounted` returns components only, even though the direct child is an element. Exmple:
```
<myComp>
  <div><otherComp /></div>
</myComp>
```
`getCurrentInstance().$children` within `onMounted` of `MyComp` will contain `OtherComp` and not the `div`. But I guess this is the way it should be - https://vuejs.org/v2/api/#vm-children - although it could be argued if `OtherComp` is a direct child component of `MyComp`.

In summary - I guess `getCurrentInstance().$children` is the official replacement for `this.$children`, right?

**Comment by antfu** [comment]
It shouldn't be `div` - they are child components instances. If you are looking for the DOM, you may want to try `vm.$el`.

**Comment by adamreisnz** [comment]
Is `getCurrentInstance().$children` still the preferred alternative?
